onnx/helper.py

```diff
@@ -533,28 +533,46 @@ def make_sequence_value_info(
     return value_info_proto
 
 
-def make_optional_value_info(
+def make_optional_tensor_value_info(
         name,  # type: Text
         elem_type,  # type: int
         shape,  # type: Optional[Sequence[Union[Text, int]]]
         doc_string="",  # type: Text
         elem_shape_denotation=None,  # type: Optional[List[Text]]
 ):  # type: (...) -> ValueInfoProto
     """Makes a ValueInfoProto based on the data type and shape for Optional."""
+    assert elem_type == OptionalProto.TENSOR
+
+    value_info_proto = ValueInfoProto()
+    value_info_proto.name = name
+    if doc_string:
+        value_info_proto.doc_string = doc_string
+
+    optional_type_proto = value_info_proto.type.optional_type
+    optional_type_proto.elem_type.tensor_type.elem_type = elem_type
+    tensor_value_info = make_tensor_value_info(name, elem_type, shape, doc_string, elem_shape_denotation)
+    if shape is not None:
+        optional_type_proto.elem_type.tensor_type.shape.CopyFrom(tensor_value_info.type.tensor_type.shape)
+    return value_info_proto
+
+
+def make_optional_sequence_value_info(
+        name,  # type: Text
+        elem_type,  # type: int
+        shape,  # type: Optional[Sequence[Union[Text, int]]]
+        doc_string="",  # type: Text
+        elem_shape_denotation=None,  # type: Optional[List[Text]]
+):  # type: (...) -> ValueInfoProto
+    """Makes a ValueInfoProto based on the data type and shape for Optional."""
+    assert elem_type == OptionalProto.SEQUENCE
```

Comment:
There seems to be some potential confusion between this and the passing of elem_type to ```make_sequence_value_info``` down below ... both elem_types cannot be the same, they must be different, right?

Reply:
Fixed this with another argument for seq_elem_type

Reply:
building sequence is a bit complicated. Theoretically sequence can contain a tensor or a sequence of tensors or a sequence of sequence of tensors etc...

Current make_sequence_value_info only support Sequence of Tensors ... Are use cases for Optional type to support anything other Sequence of Tensors? if so then make_sequence_valu_info needs to be fixed. 

Regardless we should add some documentation and checks around this

Reply:
Nit: IMHO, we can just drop the "elem_type" parameter, which is used only in the assert. We only need the seq_elem_type.